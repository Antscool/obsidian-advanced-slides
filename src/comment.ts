export type Comment = {

	type: string,
	style: string[],
	clazz: string[],
	attributes: Map<string, string>,
}

export class CommentParser {

	private readCommentRegex = /<!--.*-->/;
	private styleRegex = /(?:style="([^"]+)")|(?:style='([^']+)')/;
	private classRegex = /(?:class="([^"]+)")|(?:class='([^']+)')/;
	private parseRegex = /<!--\s*\.(element|slide):\s*(.*)-->/;
	private parsePropertiesRegex = /([^=]*)\s*=\s*"([^"]*)"\s*|([^=]*)\s*=\s*'([^']*)'\s*/g;

	commentToString(comment: Comment): string {
		return `<!-- .${comment.type}: ${this.buildAttributes(comment)} -->`;
	}

	buildAttributes(comment: Comment): string {
		return `style="${comment.style.join('; ')}" class="${comment.clazz.join(' ')}" ${this.attributeMapToString(comment.attributes)}`;
	}

	parseLine(line: string): Comment {
		return this.lineHasComment(line)
			? this.parseComment(this.readCommentStringFromLine(line))
			: null;
	}

	parseComment(comment: string): Comment {
		try {
			let [, type, properties] = this.parseRegex.exec(comment);
			let [style, clazz, attributes] = this.parseProperties(properties);
			return this.buildComment(type, style, clazz, attributes);
		} catch (ex) {
			console.log("ERROR: Cannot parse comment: " + comment);
			return null;
		}
	}

	buildComment(type: string, style: string[] = [], clazz: string[] = [], attributes: Map<string, string> = new Map<string, string>()): Comment {
		return {
			type: type,
			style: style,
			clazz: clazz,
			attributes: attributes,
		}
	}

	private attributeMapToString(attributes: Map<string, string>): string {
		let result = [];
		for (const [key, value] of attributes) {
			result.push(`${key}="${value}"`);
		}
		return result.join(" ");
	}

	private lineHasComment(line: string): boolean {
		return line?.includes("<!--") && line?.includes("-->");
	}

	private readCommentStringFromLine(line: string): string {
		return this.readCommentRegex.exec(line)?.[0] ?? '';
	}

	private parseProperties(properties: string): [string[], string[], Map<string, string>] {
		let result: [string[], string[], Map<string, string>] = [[], [], new Map<string, string>()];

		if (properties.includes('style=')) {
			result[0] = this.parseStyle(properties);
		}

		if (properties.includes('class=')) {
			result[1] = this.parseClass(properties);
		}

		result[2] = this.parseAttributes(properties);

		return result;
	}

	private parseStyle(comment: string): string[] {
		let result = this.styleRegex.exec(comment)?.[1] ?? '';
		return result.split(";").map((attr) => {
			return attr.trim();
		});
	}

	private parseClass(comment: string): string[] {
		let result = this.classRegex.exec(comment)?.[1] ?? '';
		return result.split(" ").map((attr) => {
			return attr.trim();
		});
	}

	private parseAttributes(properties: string): Map<string, string> {
		let m;
		const attributes = new Map<string, string>();

		while ((m = this.parsePropertiesRegex.exec(properties)) !== null) {
			if (m.index === this.parsePropertiesRegex.lastIndex) {
				this.parsePropertiesRegex.lastIndex++;
			}

			let key: string;
			let value: string;

			m.forEach((match, groupIndex) => {
				if (groupIndex == 1 || groupIndex == 3) {
					key = match;
				}
				if (groupIndex == 2 || groupIndex == 4) {
					value = match;
					attributes.set(key, value);
				}
			});

		}

		attributes.delete(undefined);
		attributes.delete('style');
		attributes.delete('class');
		return attributes;
	}

}
