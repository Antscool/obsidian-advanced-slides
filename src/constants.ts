export const ICON_DATA = '<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100"><path fill="currentColor" stroke="currentColor" d="M100 74.242V1.516H0v72.726h45.453v15.153H31.816v9.09h36.368v-9.09H54.547V74.242Zm-90.91-9.09V10.605h81.82v54.547Zm0 0"/><path fill="currentColor" stroke="currentColor" d="M54.547 19.695h9.09V56.06h-9.09ZM72.727 25.758h9.09v30.305h-9.09ZM36.363 31.816h9.09V56.06h-9.09ZM18.184 22.727h9.09v33.335h-9.09Zm0 0"/></svg>';
