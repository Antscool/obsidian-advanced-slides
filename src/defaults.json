{
	"theme": "black",
	"highlightTheme": "zenburn",
	"template": "template/reveal.html",
	"separator": "\r?\n---\r?\n",
	"verticalSeparator": "\r?\n--\r?\n",
	"enableLinks": false,
	"title": "",
	"css": ""
}
