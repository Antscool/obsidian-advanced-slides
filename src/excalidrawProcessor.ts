import { Notice } from "obsidian";
import { ObsidianUtils } from "./obsidianUtils";

export class ExcalidrawProcessor {
	private excalidrawImageRegex = /!\[\[(.*\.excalidraw)\|?([^\]]*)??\]\]\s?(<!--.*-->)?/i

	private utils: ObsidianUtils;

	constructor(utils: ObsidianUtils) {
		this.utils = utils;
	}

	process(markdown: string){
		return markdown
			.split('\n')
			.map((line) => {
				if (this.excalidrawImageRegex.test(line))
					return this.transformLine(line);
				return line;
			})
			.join('\n');
	}

	private transformLine(line: string) {

		var [, image, ext, comment] = this.excalidrawImageRegex.exec(line);
		var imgFile = this.utils.findImageEx(image);

		if (imgFile === null) {
			new Notice(`Cannot find Image for ${image}. Make sure to activate Auto-export SVG/PNG in Excalidraw Settings.`, 8000);
			return line;
		}
		return `![[${imgFile}${ext==undefined ? '' : '|' + ext}]] ${comment ?? ''}`;
	}
}


