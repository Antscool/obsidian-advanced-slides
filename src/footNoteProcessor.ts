
export class FootnoteProcessor {

	private regex = /\[\^([^\]]*)]/gmi;

	process(markdown: string, options: any){

		var output = markdown;

		var result = markdown
        .split(new RegExp(options.separator, 'gmi'))
        .map((slidegroup, index) => {
            return slidegroup
                .split(new RegExp(options.verticalSeparator, 'gmi'))
                .map((slide, index) => {
                    if (this.regex.test(slide)) {
						var newSlide = this.transformFootNotes(slide);
						output = output.replace(slide, newSlide);
                        return newSlide;
                    }
                    return slide;
                })
                .join(options.verticalSeparator);
        })
        .join(options.separator);
    return output;
	}

	transformFootNotes(markdown: string) {

		var input = markdown;
		var noteIdx = 1;
	
		var footNotes = new Map();
	
		var reResult : RegExpExecArray;
		while (reResult = this.regex.exec(input)) {
	
			input = input
				.split('\n')
				.map((line, index) => {
					if (line.includes(reResult[0])) {
						if (line.includes(reResult[0] + ": ")) {
							if (!footNotes.has(reResult[1])) {
								footNotes.set(reResult[1], line.split(reResult[0] + ": ")[1]);
							}
							return "";
						} else {
							var split = line.split(reResult[0]);
	
							var result = split[0].trim();
							result += '<sup id="fnref:' + reResult[1] + '" role="doc-noteref">' + noteIdx + '</sup>';
							result += '\n' + split[1].trim();
	
							noteIdx = noteIdx + 1;
	
							return result;
	
						}
					}
					return line;
				})
				.join('\n');
	
	
		}
	
		var footNotesBlock = '';
		footNotesBlock += '<div class="footnotes" role="doc-endnotes">\n';
		footNotesBlock += '<ol>\n';
	
		footNotes.forEach((value, key) => {
			footNotesBlock += '<li id="fn:' + key + '" role="doc-endnote"><p>' + value + '&nbsp;</p></li>';
		});
	
	
		footNotesBlock += '</ol>\n';
		footNotesBlock += '</div>\n';
	
		return input + '\n' + footNotesBlock;
	}

}


