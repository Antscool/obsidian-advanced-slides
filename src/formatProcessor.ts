
export class FormatProcessor {

	private regex = /==([^=]*)==/gm;
	private commentRegex = /%%([^%]*)%%/gm;

	process(markdown: string){
		return markdown.replaceAll('__','**')
		.replaceAll(this.regex,'<mark>$1</mark>')
		.replaceAll(this.commentRegex,'');
	}

}
