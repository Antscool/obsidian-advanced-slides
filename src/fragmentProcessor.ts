import { CommentParser } from "./comment";
import { Options } from "./options";

export class FragmentProcessor {

	private parser: CommentParser;
	private fragmentCounter = 1;
	private orderedListRegex = /\d\) /g;

	constructor() {
		this.parser = new CommentParser();
	}

	process(markdown: string, options: Options) {
		let output = markdown;

		markdown
			.split(new RegExp(options.separator, 'gmi'))
			.map((slidegroup) => {
				return slidegroup
					.split(new RegExp(options.verticalSeparator, 'gmi'))
					.map((slide) => {
						this.fragmentCounter = 1;

						const newSlide = slide.split('\n')
							.map((line) => {
								if (line.trim().startsWith("+ ") || this.orderedListRegex.test(line.trim())) {
									return this.transformLine(line);
								}
								return line;
							})
							.join('\n');
						output = output.split(slide).join(newSlide);
						return newSlide;

					})
					.join(options.verticalSeparator);
			})
			.join(options.separator);

		return output;
	}

	transformLine(line: string) {
		const comment = this.parser.parseLine(line) ?? this.parser.buildComment('element');

		if (line.includes('<!--')) {
			line = line.substring(0, line.indexOf('<!--'));
		}

		line = line.replaceAll("+ ", "- ");
		line = line.replaceAll(this.orderedListRegex, '1. ');

		if (!comment.attributes.get('data-fragment-index')) {
			comment.attributes.set("data-fragment-index", this.fragmentCounter.toString());
			if (!comment.clazz.includes('fragment')) {
				comment.clazz.push('fragment');
			}
			this.fragmentCounter++;
		}
		const output = line + this.parser.commentToString(comment);
		return output;
	}

}
