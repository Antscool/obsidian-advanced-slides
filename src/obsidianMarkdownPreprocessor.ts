import { App } from "obsidian";
import { BlockProcessor } from "./blockProcessor";
import { FootnoteProcessor } from "./footNoteProcessor";
import { FormatProcessor } from "./formatProcessor";
import { ImageProcessor } from "./imageProcessor";
import { InternalLinkProcessor } from "./internalLinkProcessor";
import { LatexProcessor } from "./latexProcessor";
import { MultipleFileProcessor } from "./multipleFileProcessor";

export class ObsidianMarkdownPreprocessor {

	private multipleFileProcessor : MultipleFileProcessor;
	private blockProcessor : BlockProcessor;
	private imageProcessor : ImageProcessor;
	private internalLinkProcessor : InternalLinkProcessor;
	private footnoteProcessor : FootnoteProcessor;
	private latexProcessor : LatexProcessor;
	private formatProcessor : FormatProcessor;

	constructor(app: App) {
		this.multipleFileProcessor = new MultipleFileProcessor(app);
		this.blockProcessor = new BlockProcessor();
		this.imageProcessor = new ImageProcessor(app);
		this.internalLinkProcessor = new InternalLinkProcessor();
		this.footnoteProcessor = new FootnoteProcessor();
		this.latexProcessor = new LatexProcessor();
		this.formatProcessor = new FormatProcessor();
	}

	process(markdown: string, options: any){
		const afterMultipleFileProcessor = this.multipleFileProcessor.process(markdown);
		const afterBlockProcessor = this.blockProcessor.process(afterMultipleFileProcessor);
		const afterFootNoteProcessor = this.footnoteProcessor.process(afterBlockProcessor, options);
		const afterImageProcessor = this.imageProcessor.process(afterFootNoteProcessor);
		const afterInternalLinkProcessor = this.internalLinkProcessor.process(afterImageProcessor);
		const afterLatexProcessor = this.latexProcessor.process(afterInternalLinkProcessor);
		const afterFormatProcessor = this.formatProcessor.process(afterLatexProcessor);
		return afterFormatProcessor;
	}



}


