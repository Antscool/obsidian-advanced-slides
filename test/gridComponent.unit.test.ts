import { ObsidianMarkdownPreprocessor } from "src/obsidianMarkdownPreprocessor";
import { when } from "ts-mockito";
import { prepare } from "./testUtils";
import { MockedObsidianUtils, obsidianUtils as utilsInstance} from "./__mocks__/mockObsidianUtils";


test('Grid Component > Basic Syntax', () => {

	const input =
`<grid drag="60 55" drop="5 10" style="background-color: coral;">

### Left
</grid>

<grid drag="25 55" drop="70 10" style="background-color: coral;">

### Right
</grid>

<grid drag="90 20" drop="5 70" style="background-color: coral;">

### Bottom
</grid>`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Grid Component > Basic Syntax', () => {

	const input =
`<grid drag="60 55" drop="5 10" style="background-color: coral;">

### Left
</grid>

<grid drag="25 55" drop="-5 10" style="background-color: coral;">

### Right
</grid>

<grid drag="90 20" drop="5 -10" style="background-color: coral;">

### Bottom
</grid>`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Grid Component > Position by Name', () => {

	const input =
`<grid drag="40 30" drop="topleft" style="background-color: red;">

### Top Left
</grid>

<grid drop="right" style="background-color: green;">

### Right Default Size
</grid>

<grid drag="80 30" drop="bottom" style="background-color: coral;">

### Bottom
</grid>`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Grid Component > Position Coordinates', () => {

	const input =
`<grid drag="40 50" drop="10 15" style="background-color: orange;">

### Positive X, Y
</grid>

<grid drag="30 30" drop="-40px 40px" style="background-color: red;">

### Negative X, Positive Y
</grid>

<grid drag="40 40" drop="-5 -20" style="background-color: blue;">

### Negative X, Y
</grid>

<grid drag="100 10" drop="0 -5" style="background-color: green;">

### Positive X, Negative Y
</grid>`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Grid Component > Column Flow', () => {

	when(MockedObsidianUtils.getAbsolutePath("Image.jpg.md")).thenCall( (arg) => {
		return null;
	});

	when(MockedObsidianUtils.findFile('Image.jpg')).thenCall( (arg) => {
		return '/documentation/Image.jpg';
	});

	const input =
`<grid  drag="40 100" drop="center" style="background-color: coral;" flow="col">

### Lorem
![[Image.jpg]]
### Ipsum
</grid>`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Grid Component > Row Flow', () => {

	when(MockedObsidianUtils.getAbsolutePath("Image.jpg|350.md")).thenCall( (arg) => {
		return null;
	});

	when(MockedObsidianUtils.findFile('Image.jpg')).thenCall( (arg) => {
		return '/documentation/Image.jpg';
	});

	const input =
`<grid  drag="100 40" drop="center" style="background-color: coral;" flow="row">

### Lorem
![[Image.jpg|350]]
### Ipsum
</grid>`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

