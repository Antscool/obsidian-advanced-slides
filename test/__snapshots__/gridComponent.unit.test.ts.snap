// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component > Attributes > Background 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: orange; position: fixed; left: 432px; top: 0px; height: 350px; width: 528px; \\" >

### Make
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: rgb(0,0,0); position: fixed; left: 0px; top: 350px; height: 350px; width: 528px; \\" >

### Noise
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: green; transform: rotate(-15deg); position: fixed; left: 360px; top: 280px; height: 140px; width: 240px; \\" >

### some
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Attributes > Border 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; border: thick dotted blue; position: fixed; left: 0px; top: 262.5px; height: 175px; width: 288px; \\" >

thick dotted blue
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; border: 4px solid white; position: fixed; left: 336px; top: 262.5px; height: 175px; width: 288px; \\" >

20px solid white
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; border: medium dashed red; position: fixed; left: 672px; top: 262.5px; height: 175px; width: 288px; \\" >

thick dotted blue
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Attributes > Filter 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: #B565A7; filter: blur(10px); position: fixed; left: 48px; top: 105px; height: 175px; width: 288px; \\" >

Text is too blurry
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: white; filter: grayscale(); position: fixed; left: 364.79999999999995px; top: 175px; height: 350px; width: 480px; \\" >

<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"\\" class=\\"\\" ></img></p>

</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Attributes > Padding 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: orange; box-sizing: border-box; padding: 0 50px; position: fixed; left: 0px; top: 0px; height: 350px; width: 480px; \\" >

###### Lorem Ipsum wasnt simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book here there
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: orange; box-sizing: border-box; padding: 20px; position: fixed; left: 480px; top: 350px; height: 350px; width: 480px; \\" >

<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"width: 800px\\" class=\\"\\" ></img></p>

</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Attributes > Padding 2`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: red; position: fixed; left: 48px; top: 70px; height: 385px; width: 576px; \\" >

### Fragments:
- Right<!-- .element: style=\\"\\" class=\\"fragment\\" data-fragment-index=\\"1\\" -->
- Bottom<!-- .element: style=\\"\\" class=\\"fragment\\" data-fragment-index=\\"2\\" -->
</div>

<div class=\\"reset-margin fragment\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: green; position: fixed; left: 672px; top: 70px; height: 385px; width: 240px; \\" data-fragment-index=\\"1\\">

Right Grid
</div>

<div class=\\"reset-margin fragment\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: gray; position: fixed; left: 48px; top: 490px; height: 140px; width: 864px; \\" data-fragment-index=\\"2\\">

Bottom Grid
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Attributes > Rotate 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: #B565A7; transform: rotate(-10deg); position: fixed; left: 336px; top: 0px; height: 175px; width: 288px; \\" >

Hello
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; background-color: #D65076; transform: rotate(40deg); position: fixed; left: 576px; top: 525px; height: 175px; width: 384px; \\" >

World!
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Basic Syntax 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 48px; top: 70px; height: 385px; width: 576px; background-color: coral;\\" >

### Left
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 672px; top: 70px; height: 385px; width: 240px; background-color: coral;\\" >

### Right
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 48px; top: 490px; height: 140px; width: 864px; background-color: coral;\\" >

### Bottom
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Basic Syntax 2`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 48px; top: 70px; height: 385px; width: 576px; background-color: coral;\\" >

### Left
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 672px; top: 70px; height: 385px; width: 240px; background-color: coral;\\" >

### Right
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 48px; top: 490px; height: 140px; width: 864px; background-color: coral;\\" >

### Bottom
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Column Flow 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 288px; top: 0px; height: 700px; width: 384px; background-color: coral;\\" >

### Lorem
<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"\\" class=\\"\\" ></img></p>

### Ipsum
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Position Coordinates 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 96px; top: 105px; height: 350px; width: 384px; background-color: orange;\\" >

### Positive X, Y
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 632px; top: 40px; height: 210px; width: 288px; background-color: red;\\" >

### Negative X, Positive Y
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 528px; top: 280px; height: 280px; width: 384px; background-color: blue;\\" >

### Negative X, Y
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 0px; top: 595px; height: 70px; width: 960px; background-color: green;\\" >

### Positive X, Negative Y
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Position by Name 1`] = `
"<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 0px; top: 0px; height: 210px; width: 384px; background-color: red;\\" >

### Top Left
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 480px; top: 0px; height: 700px; width: 480px; background-color: green;\\" >

### Right Default Size
</div>

<div class=\\"reset-margin\\" style=\\"display: flex; flex-direction: column; justify-content: space-evenly; position: fixed; left: 96px; top: 490px; height: 210px; width: 768px; background-color: coral;\\" >

### Bottom
</div>
<!-- .slide: class=\\"stack\\" -->"
`;

exports[`Grid Component > Row Flow 1`] = `
"<div class=\\"reset-margin flex-even\\" style=\\"display: flex; flex-direction: row; align-items: center; justify-content: space-evenly; position: fixed; left: 0px; top: 210px; height: 280px; width: 960px; background-color: coral;\\" >

### Lorem
<p style=\\"line-height: 0\\" class=\\"reset-paragraph\\" ><img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"width: 350px\\" class=\\"\\" ></img></p>

### Ipsum
</div>
<!-- .slide: class=\\"stack\\" -->"
`;
